src/one_dragon/utils/gpu_executor.py

```
@@ -1,17 +1,72 @@
+        factory: Callable[[], T],
+        providers: Sequence[str] | None,
+) -> T:
+    if should_serialize_providers(providers):
+        return run_sync(factory)
+    return factory()
+
+
+def should_serialize_session(session) -> bool:
+    try:
+        providers = session.get_providers()
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)